kv/urm.go

```diff
@@ -30,14 +80,13 @@ func filterMappingsFn(filter influxdb.UserResourceMappingFilter) func(m *influxd
 
 // FindUserResourceMappings returns a list of UserResourceMappings that match filter and the total count of matching mappings.
 func (s *Service) FindUserResourceMappings(ctx context.Context, filter influxdb.UserResourceMappingFilter, opt ...influxdb.FindOptions) ([]*influxdb.UserResourceMapping, int, error) {
-	ms := []*influxdb.UserResourceMapping{}
-	err := s.kv.View(func(tx Tx) error {
-		mappings, err := s.findUserResourceMappings(ctx, tx, filter)
-		if err != nil {
-			return err
-		}
-		ms = mappings
-		return nil
+	var (
+		err error
+		ms  []*influxdb.UserResourceMapping
+	)
+	err = s.kv.View(func(tx Tx) error {
+		ms, err = s.findUserResourceMappings(ctx, tx, filter)
```

Comment:
I go back and forth on this shadowing stuff. My initial gut reaction is that it's confusing, but I'm not sure what a better pattern is.

Reply:
Maybe passing in the thing that is being retrieved would make it better. Something like

```go
func (s *Service) FindUserResourceMappings(ctx context.Context, filter influxdb.UserResourceMappingFilter, opt ...influxdb.FindOptions)  {
    ms []*influxdb.UserResourceMapping

    err := s.kv.View(func(tx Tx) error {
        return s.findUserResourceMapping(ctx,tx,filter,&ms)
    })

    if err != nil {
        return nil, err
    }

    return ms, nil
}
```